### DISCUSSION TITLE
Kafka scaler throughing error in above 2.0 versions

### DISCUSSION THREAD

**Discussion by satheshs** [kedacore/keda#2024] [General]
we are getting the below error while connecting Apache Kafka scaler. But when we are using keda 1.5 version it's working without any issues.

ERROR   controllers.ScaledObject        Failed to ensure HPA is correctly created for ScaledObject      {"ScaledObject.Namespace": "dev-inference-worker", "ScaledObject.Name": "kafka-scaledobject", "error": "error getting scaler for trigger #0: error creating kafka client: kafka: client has run out of available brokers to talk to (Is your cluster reachable?)"}

**Comment by zroubalik** [comment]
What Kafka Broker version are you using? KEDA is internally using Sarama Kafka client, so it might this problem. 
https://github.com/Shopify/sarama#compatibility-and-api-stability

You might want to try different client version, in the most recent KEDA 2.4 we have introduced a setting for this: https://github.com/kedacore/keda/pull/1866
